packages/shopify-app-express/src/redirect-out-of-app.ts

```diff
@@ -1,13 +1,16 @@
 import {Request, Response} from 'express';
 
 import {AppConfigInterface} from './config-types';
-import {RedirectOutOfAppFunction, RedirectOutOfAppParams} from './types';
+import {RedirectOutOfAppFunction, ApiAndConfigParams} from './types';
 
 export function redirectOutOfApp({
+  api,
   config,
-}: RedirectOutOfAppParams): RedirectOutOfAppFunction {
+}: ApiAndConfigParams): RedirectOutOfAppFunction {
   return function redirectOutOfApp({req, res, redirectUri, shop}): void {
-    if (req.headers.authorization?.match(/Bearer (.*)/)) {
+    if (!api.config.isEmbeddedApp && isFetchRequest(req)) {
+      appBridgeHeaderRedirect(config, res, redirectUri);
```

Comment:
Since this is doing the same thing as the branch below, could we merge them with an `||`? Or would you prefer to keep them separate for clarity?